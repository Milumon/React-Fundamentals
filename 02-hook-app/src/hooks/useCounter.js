import { useState } from "react"

export const useCounter = ({ numero = 10 }) => {

    const [counter, setCounter] = useState(numero)
     
    const increment = () => {
        setCounter(counter + 1)
    }

    const decrement = () => {
        setCounter(counter - 1)

    }

    const reset = () => {
        setCounter(numero)
    }

    return {
        counter,
        increment,
        decrement,
        reset
    }
}
