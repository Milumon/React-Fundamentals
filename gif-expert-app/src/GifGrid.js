import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGifs();
    }, [])
    

    const getGifs = async () => {
        const url = 'https://api.giphy.com/v1/gifs/search?api_key=iXsYNn6eQSpxVoN6YfKSRVojGM1eOS5e&q=GFRIEND&limit=10'
        const resp = await fetch(url)
        const { data } = await resp.json() 
        const gifs = data.map( img => { 
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
         setImages(gifs)
    } 

    return (
        <>
        <h3> {category} </h3>
        <div  className='card-grid'>  
               {images.map((image) => 

                    <GifGridItem 
                    { ...image }
                    key={image.id} />
                )}  
        </div>
         </>
    )
}
