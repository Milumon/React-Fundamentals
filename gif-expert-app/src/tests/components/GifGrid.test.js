import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')


describe('El component se debe mostrar', () => {

    
    const category = 'GFRIEND' 

    test('El componente debe coincidir con el snapshot', () => { 
    
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
 
        const wrapper = shallow( <GifGrid category={category} /> )
        expect(wrapper).toMatchSnapshot()
        
    })

    test('Debe de mostrar items cuando se cargan las imagenes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'http://localhost/cualquiercosa.jpg',
            title: 'Mi imagensita'
        },{
            id: 'xd',
            url: 'http://localhost/cualquiercosa.jpg',
            title: 'Mi imagensita'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow( <GifGrid category={category} /> )
        
        // tests
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

    
    })
    
    
    
});