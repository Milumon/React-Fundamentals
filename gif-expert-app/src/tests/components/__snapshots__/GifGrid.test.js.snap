// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`El component se debe mostrar El componente debe coincidir con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    GFRIEND
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    'Carganding' 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
